pkgs/by-name/tu/turnon/package.nix

```diff
@@ -0,0 +1,49 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  cairo,
+  pango,
+  pkg-config,
+  libadwaita,
+  blueprint-compiler,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "turnon";
+  version = "0.4.0";
+
+  src = fetchFromGitHub {
+    owner = "swsnr";
+    repo = "turnon";
+    rev = "v${version}";
+    hash = "sha256-TF6mGETX6cK0ob2LLrjneifwSf+9EEjrv4x7AfWIamQ=";
+  };
+
+  cargoHash = "sha256-EpwgvaunShETZqH7v3Dux60pGLeBOtBZhpbDi7Eows4=";
+
+  nativeBuildInputs = [
+    cairo
+    pango
+    pkg-config
+    blueprint-compiler
+  ];
+
+  buildInputs = [
+    libadwaita
+  ];
+
+  strictDeps = true;
+
+  preBuild = ''
+    blueprint-compiler format resources/**/*.blp
+  '';
+
+  meta = {
+    description = "Turn on devices in your local network";
+    homepage = "https://github.com/swsnr/turnon";
+    license = lib.licenses.mpl20;
+    maintainers = with lib.maintainers; [ mksafavi ];
+    #mainProgram = "";
```

Comment:
Not needed?

Reply:
What's the common practice for mainProgram? is it used to mark a single executable as the main app for packages that have multiple executable files? also is it possible to mark multiple files as the mainProgram?

In this case there's only /bin/turnon so I think it's not needed. I'll remove it.